src/Vector.cpp

```diff
@@ -0,0 +1,119 @@
+
+//
+// Created by Dvir Avraham on 19/03/2023.
+//
+
+#include "Vector.h"
+
+// implemnt vector
+
+// constructor with size and value
+Vector::Vector(size_t size, int value) {
+    _data = new int[size];
+    _size = size;
+    _capacity = size;
+    for (int i = 0; i < size; i++) {
+        _data[i] = value;
+    }
+}
+
+// destructor
+Vector::~Vector() {
+    delete[] _data;
+}
+
+//
+
+Vector::Vector(const Vector &other) {
+    _data = new int[other._size];
+    _size = other._size;
+    _capacity = other._size;
+    for (int i = 0; i < _size; i++) {
+        _data[i] = other._data[i];
+    }
+}
+
+Vector &Vector::operator=(const Vector &other) {
+    if (this != &other) {
+        delete[] _data;
+        _data = new int[other._size];
+        _size = other._size;
+        _capacity = other._size;
+        for (int i = 0; i < _size; i++) {
+            _data[i] = other._data[i];
+        }
+    }
+    return *this;
+}
+
+int &Vector::operator[](int index) {
+    if(index < 0 || index >= _size)
+        throw std::out_of_range("index out of range");
+    return _data[index];
+}
+
+int Vector::operator[](int index) const {
+    if(index < 0 || index >= _size)
```

Comment:
And because index need to be size_t it's an unsigned integer type, so no need to check if it's smaller than 0